src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,259 @@
+        }
+    }
+
+    private static void getNewInput() throws DukeException, IOException {
+        String inputCommand = SCANNER.nextLine();
+        executeCommand(inputCommand);
+    }
+
+    private static void executeCommand(String inputCommand) throws DukeException, IOException {
+        String[] commandArr = inputCommand.split(" ");
+        int listNum;
+        switch (commandArr[0]) {
+        case COMMAND_LIST:
+            listNum = 1;
+            if (itemNum == 0) {
+                System.out.println("List is empty");
+            }
+            else {
+                for (Task a : taskArrayList) {
+                    System.out.println(listNum++ + ". " + a);
+                }
+            }
+            break;
+        case COMMAND_MARK:
+            int markNum = Integer.parseInt(commandArr[1]) - 1;
+            try {
+                taskArrayList.get(markNum).setDone(true);
+                System.out.println("Nice! I've marked this task as done:\n" + taskArrayList.get(markNum));
+            } catch(IndexOutOfBoundsException markOutOfBounds) {
+                System.out.println("Invalid mark index!!");
+            }
+            break;
+        case COMMAND_UNMARK:
+            int unMarkNum = Integer.parseInt(commandArr[1]) - 1;
+            try {
+                taskArrayList.get(unMarkNum).setDone(false);
+                System.out.println("Nice! I've marked this task as not done yet:\n" + taskArrayList.get(unMarkNum));
+            } catch(IndexOutOfBoundsException unMarkOutOfBounds) {
+                System.out.println("Unmark index is out of bounds!!");
+            }
+            break;
+        case COMMAND_DELETE:
+            int deleteIndex = Integer.parseInt(commandArr[1]) - 1;
+            try {
+                itemNum--;
+                System.out.println("Noted. I've removed this task:\n" + taskArrayList.get(deleteIndex) +
+                        "\nNow you have " + itemNum + " tasks in the list.");
+                taskArrayList.remove(deleteIndex);
+            } catch(IndexOutOfBoundsException deleteIndexOutOFBounds) {
+                System.out.println("The item you want to delete is not in the list!");
+            }
+            break;
+        case COMMAND_BYE:
+            exitProgram();
+            break;
+        default:
+            addTask(inputCommand);
+            break;
+        }
+    }
+
+    private static void saveTaskList() throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task a : taskArrayList) {
+            fw.write(String.valueOf(a.getTaskType()) + "-" + String.valueOf(a.isDone()) + "-"
+                    + a.saveDescription() + a.saveAdditionalInfo() + System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    private static void exitProgram() {
+        System.out.println("Bye. Hope to see you again soon!");
+        System.exit(0);
+    }
+
+    private static void addTask(String inCommand) throws DukeException {
```

Comment:
This is another long method (>30 LoC) which you can refactor and ensure SLAP to increase code readability.